### PR TITLE
Treatment Related Effect

### PR DESCRIPTION
**Pull request by casey1173**
Creates treatment related effect for endpoint groups.
- Adds treatment effect to endpoint group model
- Can be added via endpoint update form
- Displays on endpoint/:id/ detail page
- Updated help text for significance level
- Adds column to exports

image

Results are propagated throughout the application, and can be used in data pivots. We recommend using a white-filled green up triangle, or a white-filled red down triangle to indicate treatment-related effects:

Screen Shot 2022-08-23 at 10 22 06 AM

### CODE REVIEW

File: hawc/apps/common/templates/common/helptext_popup_js.html [hunk 1]
```diff
@@ -0,0 +1,9 @@
+<script type="text/javascript">
```
Comment (shapiromatron) [inline hawc/apps/common/templates/common/helptext_popup_js.html:1]: this is a nice pattern

File: frontend/animal/EndpointTable.js [hunk 2]
```diff
@@ -32,11 +32,18 @@ class EndpointTable {
...
+            .value();
+    }
+
+    hasData(val) {
         return _.chain(this.endpoint.data.groups)
             .map(function(d) {
                 return d[val];
             })
-            .some($.isNumeric)
+            .some(x => x != null)
```
Comment (shapiromatron) [inline frontend/animal/EndpointTable.js:46]: @casey1173 curious - why did this line change?

Comment (shapiromatron) [reply frontend/animal/EndpointTable.js:46]: can we just delete `hasData` and use `hasValues` everywhere?

**Review by shapiromatron — COMMENTED** [review]
almost there, one minor question regarding `hasValues` vs `hasData`

File: frontend/animal/EndpointTable.js (hunk 2 above)
Comment (casey1173) [reply frontend/animal/EndpointTable.js:46]: hasValues checks if all string values are numeric, and treatment related effect values are 'yes' or 'no'

Comment (shapiromatron) [reply frontend/animal/EndpointTable.js:46]: can we change to use `!_.isNil` and get rid of the duplicative function? https://lodash.com/docs/4.17.15#isNil

**Review by shapiromatron — APPROVED** [review]

**Pull request merged**